python/src/ecole/environment.py

```diff
@@ -170,9 +176,16 @@ def step(self, action, *dynamics_args, **dynamics_kwargs):
             done, action_set = self.dynamics.step_dynamics(
                 self.model, action, *dynamics_args, **dynamics_kwargs
             )
-            observation = self.observation_function.extract(self.model, done)
+
+            limit_values = self.model.pause_limits()
+            if not done:
+                observation = self.observation_function.extract(self.model, done)
+            else:
+                observation = None
             reward = self.reward_function.extract(self.model, done)
             information = self.information_function.extract(self.model, done)
+            self.model.set_params(limit_values)
```

Comment:
I would also do a separate PR to discuss this pause limits / restore limits thing. I can see a use case (the competition actually), where a reward function uses the "limits/solvingtime" parameter. I think it would be best to have each observation / reward / information function decide which parameters it wants to set / restore. At least, we should discuss it further.

Reply:
Moved to separate PR.